lib/http2/hpack.c

```diff
@@ -306,11 +376,23 @@ static int decode_header(h2o_mem_pool_t *pool, struct st_h2o_decode_header_resul
     } else {
         /* non-existing name */
         const h2o_token_t *name_token;
-        if ((result->name = decode_string(pool, src, src_end)) == NULL)
+        uint8_t flags = 0;
+        if ((result->name = decode_string(pool, src, src_end, &flags)) == NULL)
```

Comment:
How about  changeing arguments of `decode_string` to `h2o_mem_pool_t *pool, const uint8_t **src, const uint8_t *src_end, int is_header_name, const char **err_desc)`?

By changing as such, I believe the code would become simpler since you can do all the checks in `decode_string` function without using `H2O_HUFF_ENCODED` flag.